paddle/fluid/operators/math/blas_impl.h

```diff
@@ -563,6 +563,35 @@ void Blas<platform::CPUDeviceContext>::BatchedGEMM(
 #endif
 }
 
+template <typename DeviceContext>
+template <typename T>
+void Blas<DeviceContext>::BatchedGEMM(
+    CBLAS_TRANSPOSE transA, CBLAS_TRANSPOSE transB, int M, int N, int K,
+    T alpha, std::vector<const T *> *a_array, const int lda,
+    std::vector<const T *> *b_array, const int ldb, T beta,
+    std::vector<T *> *c_array, const int ldc, int batchCount) const {
+  PADDLE_ENFORCE(a_array != nullptr && a_array->size() == (size_t)batchCount);
+  PADDLE_ENFORCE(b_array != nullptr && b_array->size() == (size_t)batchCount);
+  PADDLE_ENFORCE(c_array != nullptr && c_array->size() == (size_t)batchCount);
+
+  if (std::is_same<DeviceContext, platform::CPUDeviceContext>::value) {
+#ifdef PADDLE_WITH_MKLML
+    CBlas<T>::GEMM_BATCH(CblasRowMajor, &transA, &transB, &M, &N, &K, &alpha,
+                         a_array->data(), &lda, b_array->data(), &ldb, &beta,
+                         c_array->data(), &ldc, 1 /* group_count */,
+                         &batchCount);
+    return;
+#endif
+  }
+  for (int k = 0; k < batchCount; ++k) {
+    auto *Ak = (*a_array)[k];
+    auto *Bk = (*b_array)[k];
+    auto *Ck = (*c_array)[k];
+    this->template GEMM<T>(transA == CblasTrans, transB == CblasTrans, M, N, K,
+                           alpha, Ak, lda, Bk, ldb, beta, Ck, ldc);
+  }
+}
```

Comment:
> * What's the difference between old `BatchedGEMM` and new `BatchedGEMM` in your PR?
> * I see the difference is the input format from `const T` to `std::vector<const T *> *a_array`.
According to the transpose dimensions's difference and the stride's requirement,  the array calculation of MKL BatchedGEMM need be get though the special calculation. So it tends to move the calculation into the internal of  matmul operator.
> * Could you reuse the old one or unify them? Or why do you create a new one?
The initial idea is that it can avoid the completion into the common blas's implementation.  If we need implement this or others  array's calculation into blas_impl.h, it can not keep the code's clean.
> 
> Same for the `Matmul`.
> There is already
> 
> ```
> void Blas<DeviceContext>::MatMul(const framework::Tensor &mat_a,
>                                  const MatDescriptor &dim_a,
>                                  const framework::Tensor &mat_b,
>                                  const MatDescriptor &dim_b, T alpha,
>                                  framework::Tensor *mat_out, T beta) 
> ```